src/resource_dasm.cc

```diff
@@ -2551,414 +2551,424 @@ static uint32_t parse_cli_type(const char* str, char end_char = '\0', size_t* nu
-          if (!rf.remove(op.res_type, op.res_id)) {
-            throw runtime_error("cannot delete resource");
-          }
-          fprintf(stderr, "... (delete) %s:%hd OK\n", type_str.c_str(), op.res_id);
-          break;
-        case ModificationOperation::Type::CHANGE_ID:
-          if (!rf.change_id(op.res_type, op.res_id, op.new_res_id)) {
-            throw runtime_error("cannot change resource id");
-          }
-          fprintf(stderr, "... (change id) %s:%hd=>%hd OK\n", type_str.c_str(),
-              op.res_id, op.new_res_id);
-          break;
-        case ModificationOperation::Type::RENAME:
-          if (!rf.rename(op.res_type, op.res_id, op.res_name)) {
-            throw runtime_error("cannot rename resource");
-          }
-          fprintf(stderr, "... (rename) %s:%hd=>\"%s\" OK\n", type_str.c_str(),
-              op.res_id, op.res_name.c_str());
-          break;
-        default:
-          throw logic_error("invalid modification operation");
       }
-    }
 
-    if (!use_output_data_fork) {
-      out_dir += RESOURCE_FORK_FILENAME_SUFFIX;
-    }
+      if (!use_output_data_fork) {
+        out_dir += RESOURCE_FORK_FILENAME_SUFFIX;
+      }
 
-    string output_data = serialize_resource_fork(rf);
-    fprintf(stderr, "... (serialize output) %zu bytes\n", output_data.size());
+      string output_data = serialize_resource_fork(rf);
+      fprintf(stderr, "... (serialize output) %zu bytes\n", output_data.size());
 
-    // Attempting to open the resource fork of a nonexistent file will fail
-    // without creating the file, so if we're writing to a resource fork, we
-    // touch the file first to make sure it will exist when we write the output.
-    if (ends_with(out_dir, RESOURCE_FORK_FILENAME_SUFFIX)) {
-      fopen_unique(out_dir.substr(0, out_dir.size() - RESOURCE_FORK_FILENAME_SUFFIX.size()), "a+");
-    } else if (ends_with(out_dir, RESOURCE_FORK_FILENAME_SHORT_SUFFIX)) {
-      fopen_unique(out_dir.substr(0, out_dir.size() - RESOURCE_FORK_FILENAME_SHORT_SUFFIX.size()), "a+");
-    }
-    save_file(out_dir, output_data);
+      // Attempting to open the resource fork of a nonexistent file will fail
+      // without creating the file, so if we're writing to a resource fork, we
+      // touch the file first to make sure it will exist when we write the output.
+      if (ends_with(out_dir, RESOURCE_FORK_FILENAME_SUFFIX)) {
+        fopen_unique(out_dir.substr(0, out_dir.size() - RESOURCE_FORK_FILENAME_SUFFIX.size()), "a+");
+      } else if (ends_with(out_dir, RESOURCE_FORK_FILENAME_SHORT_SUFFIX)) {
+        fopen_unique(out_dir.substr(0, out_dir.size() - RESOURCE_FORK_FILENAME_SHORT_SUFFIX.size()), "a+");
+      }
+      save_file(out_dir, output_data);
 
-    return 0;
+      return 0;
+    }
+  }
+  catch (const std::exception& e) {
+    fprintf(stderr, "Error: %s\n", e.what());
+    return 1;
+  }
+  catch (...) {
+    fprintf(stderr, "An unknown error occurred\n");
+    return 1;

```

Comment:
We should never throw something that isn't a std::exception, so this should never occur. If it does, I think we should actually crash.

Reply:
Good point. Fixed.